### DISCUSSION TITLE
Script: FlashFS and SD some issues

### DISCUSSION THREAD

**Discussion by nikito7** [arendst/Tasmota#22039] [General]
Hardware:
- ESP32 C6 (maybe S3 too)
- Micro SD Card

Software version:
- self compiled: 
Program Version	14.2.0.3 (tasmota32)
Build Date & Time	2024-08-27T20:32:13
Core/SDK Version	3_0_4/5.1.4.240801

Issue 1:

res=fz(fr) overflows if file do not exist.
in esp8266 I used fz(fr) to check if file exist.

```
18:05:00.744 CSV: Headers LP-2024-08.csv **4278124288.00**
18:05:00.746 CSV: Updating LP-2024-08.csv 32.00
18:05:01.876 CFG: Saved, Count 410, Bytes 4096
```

The overflow it's random **4278124288.00**

Issue 2:

UfsRun only run files from FlashFS

**Comment by gemu2015** [comment]
this checks if a file exists:
fx("fname") check if file fname exists

fz() returns the file size of an open file
you must check the value of fr=fo() (file open)  before doing something with fr ( fr >= 0)

TFS is fixed to use flash file system only,

in scripter you may precede the path with 
/ffs/ for flash file access
/sdfs/ for sd card access
default is flash if sd card not found, else sd card
